pep/app/pods/browse/read/controller.ts

```diff
@@ -89,6 +89,23 @@ export default class BrowseRead extends Controller {
         return this.document?.PEPCode === IJP_OPEN_CODE;
     }
 
+    /**
+     * Sets up the session authenticated to refresh the document
+     *
+     * @memberof SearchRead
+     */
+    setup() {
+        this.session.on('authenticationAndSetupSucceeded', this.onAuthenticated);
+    }
+
+    /**
+     * Removes the auth succeeded event listener on destroy of controller
+     */
+    willDestroy() {
+        super.willDestroy();
+        this.session.off('authenticationAndSetupSucceeded', this.onAuthenticated);
+    }
```

Comment:
since controllers are singletons they shouldn't normally be destroyed over the life of the app right? If thats the case, I wonder if you need to remove the listener on like the `didTransition` of the route or something too, otherwise you might be setting up multiple duplicate listeners each time the route is visited.

Reply:
Yes good point, honestly should use 'resetController' on the route